Thrift bool parameter gets into unknown volo_genbool type, methods started with '_' losts underscore symbol

## Bug Report

Encountered compilation errors when trying to compile a simple Thrift IDL using the volo crate. The errors include unknown types and incorrect handling of underscored method names, which is critical for backward compatibility with existing service definitions.

### Version
```
cargo tree | grep -E "volo|pilota"
volo_test v0.1.0 (/home/alex/git/volo-test)
├── pilota v0.11.7
├── volo v0.10.3
├── volo-gen v0.1.0 (/home/alex/git/volo-test/volo-gen)
│   ├── pilota v0.11.7 (*)
│   ├── volo v0.10.3 (*)
│   └── volo-thrift v0.10.4
│       ├── pilota v0.11.7 (*)
│       └── volo v0.10.3 (*)
│   └── volo-build v0.10.15
│       ├── pilota-build v0.11.22
│       │   ├── pilota-thrift-parser v0.11.2
│       ├── volo v0.10.3 (*)
└── volo-thrift v0.10.4 (*)
```

### Platform
```
uname -a
Linux P53 5.15.0-91-generic #101-Ubuntu SMP Tue Nov 14 13:30:08 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux
```
### Crates

`volo, pilota`

### Description

#### Steps to Reproduce:

##### Create a Sample IDL File:

```
~/git/volo-test> tree
.
└── rts
    └── test.thrift

~/git/volo-test> cat rts/test.thrift
service Test {
    string health(
        1: bool deep_check
    )

    string _underscored(
        1: string param
    )
}
```

##### Initialize a Volo Project:
```
~/git/volo-test> volo init volo-test rts/test.thrift`
cargo:rerun-if-changed=/home/alex/git/volo-test/rts/test.thrift
```

##### Resulting directory structure:
```
~/git/volo-test [master|…6]> tree
.
├── Cargo.lock
├── Cargo.toml
├── rts
│   └── test.thrift
├── src
│   ├── bin
│   │   └── server.rs
│   └── lib.rs
└── volo-gen
    ├── build.rs
    ├── Cargo.toml
    ├── src
    │   └── lib.rs
    └── volo.yml
```

#### Review Generated Code:
In src/lib.rs, observe issues such as:

* Unknown type volo_genbool.
* Incorrect method naming leading to _underscored issues.
```
~/git/volo-test [master|…5]> cat src/lib.rs
pub struct S;

impl volo_gen::test::Test for S {
    async fn health(
        &self,
        _deep_check: volo_genbool,  // <<< this is unknown
    ) -> ::core::result::Result<::pilota::FastStr, ::volo_thrift::ServerError> {
        ::std::result::Result::Ok(Default::default())
    }
    async fn underscored(  // <<< underscore is missing
        &self,
        _param: volo_gen::pilota::FastStr,
    ) -> ::core::result::Result<::pilota::FastStr, ::volo_thrift::ServerError> {
        ::std::result::Result::Ok(Default::default())
    }
}
```

#### Build the Project:
```
cargo build
...
error[E0422]: cannot find struct, variant or union type `Test_UnderscoredArgsSend` in this scope
    --> /home/alex/git/volo-test/target/debug/build/volo-gen-47336832199418a2/out/volo_gen.rs:1112:57
     |
144  |         pub struct TestUnderscoredArgsSend {
     |         ---------------------------------- similarly named struct `TestUnderscoredArgsSend` defined here
...
1112 |                 let req = TestRequestSend::_Underscored(Test_UnderscoredArgsSend { param });
     |                                                         ^^^^^^^^^^^^^^^^^^^^^^^^ help: a struct with a similar name exists: `TestUnderscoredArgsSend`

error[E0422]: cannot find struct, variant or union type `Test_UnderscoredArgsSend` in this scope
    --> /home/alex/git/volo-test/target/debug/build/volo-gen-47336832199418a2/out/volo_gen.rs:1169:57
     |
144  |         pub struct TestUnderscoredArgsSend {
     |         ---------------------------------- similarly named struct `TestUnderscoredArgsSend` defined here
...
1169 |                 let req = TestRequestSend::_Underscored(Test_UnderscoredArgsSend { param });
     |                                                         ^^^^^^^^^^^^^^^^^^^^^^^^ help: a struct with a similar name exists: `TestUnderscoredArgsSend`

error[E0412]: cannot find type `Test_UnderscoredArgsRecv` in this scope
    --> /home/alex/git/volo-test/target/debug/build/volo-gen-47336832199418a2/out/volo_gen.rs:1280:26
     |
447  |         pub struct TestUnderscoredArgsRecv {
     |         ---------------------------------- similarly named struct `TestUnderscoredArgsRecv` defined here
...
1280 |             _Underscored(Test_UnderscoredArgsRecv),
     |                          ^^^^^^^^^^^^^^^^^^^^^^^^ help: a struct with a similar name exists: `TestUnderscoredArgsRecv`

error[E0412]: cannot find type `Test_UnderscoredArgsSend` in this scope
    --> /home/alex/git/volo-test/target/debug/build/volo-gen-47336832199418a2/out/volo_gen.rs:1286:26
     |
144  |         pub struct TestUnderscoredArgsSend {
     |         ---------------------------------- similarly named struct `TestUnderscoredArgsSend` defined here
...
1286 |             _Underscored(Test_UnderscoredArgsSend),
     |                          ^^^^^^^^^^^^^^^^^^^^^^^^ help: a struct with a similar name exists: `TestUnderscoredArgsSend`

```

Decoder part from out/volo_gen.rs still has an underscore in name:
```
            fn decode<T: ::pilota::thrift::TInputProtocol>(
                __protocol: &mut T,
                msg_ident: &::pilota::thrift::TMessageIdentifier,
            ) -> ::core::result::Result<Self, ::pilota::thrift::ThriftException> {
                ::std::result::Result::Ok(match &*msg_ident.name {
                    "health" => Self::Health(::pilota::thrift::Message::decode(__protocol)?),

                    "_underscored" => { // <<< underscore is still here!

                        Self::_Underscored(::pilota::thrift::Message::decode(__protocol)?)
                    }
                    _ => {
                        return ::std::result::Result::Err(
                            ::pilota::thrift::new_application_exception(
                                ::pilota::thrift::ApplicationExceptionKind::UNKNOWN_METHOD,
                                format!("unknown method {}", msg_ident.name),
                            ),
                        );
                    }
                })
            }
```

After removing `_` from IDL method definition we can see another error
```
~/git/volo-test [master|…6]> cargo build
   Compiling volo-gen v0.1.0 (/home/alex/git/volo-test/volo-gen)
   Compiling volo_test v0.1.0 (/home/alex/git/volo-test)
error[E0433]: failed to resolve: could not find `pilota` in `volo_gen`
  --> src/lib.rs:12:27
   |
12 |         _param: volo_gen::pilota::FastStr,
   |                           ^^^^^^ could not find `pilota` in `volo_gen`

error[E0412]: cannot find type `volo_genbool` in this scope
 --> src/lib.rs:6:22
  |
6 |         _deep_check: volo_genbool,
  |                      ^^^^^^^^^^^^ not found in this scope
```

The ability to correctly parse and handle IDLs with underscored method names is crucial for maintaining compatibility with existing production services. Adjustments to Thrift IDLs for naming conventions without underscores are not feasible due to service dependencies.

If further context or any specific file changes are required, please let me know so I can assist with providing them.

Sorry to bother you again, I tried to compile our company thrift IDLs and now see other issues:
```
# some subset of definitions
struct TSchemaField {
    1: required string name;
    6: optional map<string, list<string>> selectors;
}

service Test {
    string health(
        1: bool deep_check
    )
    string test_map(
        1: required string name;
        6: optional set<TSchemaField> fields;
    )
}
```

```
cargo tree | grep -E "volo|pilota"
volo_test v0.1.0 (/home/alex/git/volo-test)
├── pilota v0.11.7
├── volo v0.10.3
├── volo-gen v0.1.0 (/home/alex/git/volo-test/volo-gen)
│   ├── pilota v0.11.7 (*)
│   ├── volo v0.10.3 (*)
│   └── volo-thrift v0.10.4
│       ├── pilota v0.11.7 (*)
│       └── volo v0.10.3 (*)
│   └── volo-build v0.10.16
│       ├── pilota-build v0.11.23
│       │   ├── pilota-thrift-parser v0.11.3
│       ├── volo v0.10.3 (*)
└── volo-thrift v0.10.4 (*)

```

```
    --> /home/alex/git/volo-test/target/debug/build/volo-gen-470dfe4d935de4fb/out/volo_gen.rs:1077:13
     |
1073 |         #[derive(Debug, Default, Clone, PartialEq)]
     |                                         --------- in this derive macro expansion
...
1077 |             pub fields: ::pilota::AHashSet<TSchemaField>,
     |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
     |
note: the following type would have to `impl` its required traits for this operation to be valid
    --> /home/alex/git/volo-test/target/debug/build/volo-gen-470dfe4d935de4fb/out/volo_gen.rs:6:9
     |
6    |         pub struct TSchemaField {
     |         ^^^^^^^^^^^^^^^^^^^^^^^
     |         |
     |         must implement `Hash`
     |         must implement `std::cmp::Eq`
     = note: this error originates in the derive macro `PartialEq` (in Nightly builds, run with -Z macro-backtrace for more info)
help: consider annotating `TSchemaField` with `#[derive(Eq, Hash, PartialEq)]`
     |
6    +         #[derive(Eq, Hash, PartialEq)]
7    |         pub struct TSchemaField {
     |

error[E0277]: the trait bound `TSchemaField: std::cmp::Eq` is not satisfied
    --> /home/alex/git/volo-test/target/debug/build/volo-gen-470dfe4d935de4fb/out/volo_gen.rs:1140:45
     |
1140 | ...                   val.insert(::pilota::thrift::Message::decode(__protocol)?);
     |                           ^^^^^^ the trait `std::cmp::Eq` is not implemented for `TSchemaField`
     |
note: required by a bound in `HashSet::<T, S>::insert`
    --> /home/alex/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/std/src/collections/hash/set.rs:429:8
     |
429  |     T: Eq + Hash,
     |        ^^ required by this bound in `HashSet::<T, S>::insert`
...
889  |     pub fn insert(&mut self, value: T) -> bool {
     |            ------ required by a bound in this associated function
help: consider annotating `TSchemaField` with `#[derive(Eq)]`
     |
6    +         #[derive(Eq)]
7    |         pub struct TSchemaField {
     |

error[E0277]: the trait bound `TSchemaField: Hash` is not satisfied
    --> /home/alex/git/volo-test/target/debug/build/volo-gen-470dfe4d935de4fb/out/volo_gen.rs:1140:45
     |
1140 | ...                   val.insert(::pilota::thrift::Message::decode(__protocol)?);
     |                           ^^^^^^ the trait `Hash` is not implemented for `TSchemaField`
     |
note: required by a bound in `HashSet::<T, S>::insert`
    --> /home/alex/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/std/src/collections/hash/set.rs:429:13
     |
429  |     T: Eq + Hash,
     |             ^^^^ required by this bound in `HashSet::<T, S>::insert`
...
889  |     pub fn insert(&mut self, value: T) -> bool {
     |            ------ required by a bound in this associated function

error[E0277]: the trait bound `TSchemaField: std::cmp::Eq` is not satisfied
    --> /home/alex/git/volo-test/target/debug/build/volo-gen-470dfe4d935de4fb/out/volo_gen.rs:1223:29
     |
1223 |                         val.insert(<TSchemaField as ::pilota::thrift::Message>::decode_async(__protocol).await?);
     |                             ^^^^^^ the trait `std::cmp::Eq` is not implemented for `TSchemaField`
     |
note: required by a bound in `HashSet::<T, S>::insert`
    --> /home/alex/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/std/src/collections/hash/set.rs:429:8
     |
429  |     T: Eq + Hash,
     |        ^^ required by this bound in `HashSet::<T, S>::insert`
...
889  |     pub fn insert(&mut self, value: T) -> bool {
     |            ------ required by a bound in this associated function
help: consider annotating `TSchemaField` with `#[derive(Eq)]`
     |
6    +         #[derive(Eq)]
7    |         pub struct TSchemaField {
     |

error[E0277]: the trait bound `TSchemaField: Hash` is not satisfied
    --> /home/alex/git/volo-test/target/debug/build/volo-gen-470dfe4d935de4fb/out/volo_gen.rs:1223:29
     |
1223 |                         val.insert(<TSchemaField as ::pilota::thrift::Message>::decode_async(__protocol).await?);
     |                             ^^^^^^ the trait `Hash` is not implemented for `TSchemaField`
     |
note: required by a bound in `HashSet::<T, S>::insert`
    --> /home/alex/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/std/src/collections/hash/set.rs:429:13
     |
429  |     T: Eq + Hash,
     |             ^^^^ required by this bound in `HashSet::<T, S>::insert`
...
889  |     pub fn insert(&mut self, value: T) -> bool {
     |            ------ required by a bound in this associated function

error[E0369]: binary operation `==` cannot be applied to type `AHashSet<TSchemaField>`
    --> /home/alex/git/volo-test/target/debug/build/volo-gen-470dfe4d935de4fb/out/volo_gen.rs:1446:13
     |
1442 |         #[derive(Debug, Default, Clone, PartialEq)]
     |                                         --------- in this derive macro expansion
...
1446 |             pub fields: ::pilota::AHashSet<TSchemaField>,
     |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
     |
note: the following type would have to `impl` its required traits for this operation to be valid
    --> /home/alex/git/volo-test/target/debug/build/volo-gen-470dfe4d935de4fb/out/volo_gen.rs:6:9
     |
6    |         pub struct TSchemaField {
     |         ^^^^^^^^^^^^^^^^^^^^^^^
     |         |
     |         must implement `Hash`
     |         must implement `std::cmp::Eq`
     = note: this error originates in the derive macro `PartialEq` (in Nightly builds, run with -Z macro-backtrace for more info)
help: consider annotating `TSchemaField` with `#[derive(Eq, Hash, PartialEq)]`
     |
6    +         #[derive(Eq, Hash, PartialEq)]
7    |         pub struct TSchemaField {
     |

error[E0277]: the trait bound `TSchemaField: std::cmp::Eq` is not satisfied
    --> /home/alex/git/volo-test/target/debug/build/volo-gen-470dfe4d935de4fb/out/volo_gen.rs:1509:45
     |
1509 | ...                   val.insert(::pilota::thrift::Message::decode(__protocol)?);
     |                           ^^^^^^ the trait `std::cmp::Eq` is not implemented for `TSchemaField`
     |
note: required by a bound in `HashSet::<T, S>::insert`
    --> /home/alex/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/std/src/collections/hash/set.rs:429:8
     |
429  |     T: Eq + Hash,
     |        ^^ required by this bound in `HashSet::<T, S>::insert`
...
889  |     pub fn insert(&mut self, value: T) -> bool {
     |            ------ required by a bound in this associated function
help: consider annotating `TSchemaField` with `#[derive(Eq)]`
     |
6    +         #[derive(Eq)]
7    |         pub struct TSchemaField {
     |

error[E0277]: the trait bound `TSchemaField: Hash` is not satisfied
    --> /home/alex/git/volo-test/target/debug/build/volo-gen-470dfe4d935de4fb/out/volo_gen.rs:1509:45
     |
1509 | ...                   val.insert(::pilota::thrift::Message::decode(__protocol)?);
     |                           ^^^^^^ the trait `Hash` is not implemented for `TSchemaField`
     |
note: required by a bound in `HashSet::<T, S>::insert`
    --> /home/alex/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/std/src/collections/hash/set.rs:429:13
     |
429  |     T: Eq + Hash,
     |             ^^^^ required by this bound in `HashSet::<T, S>::insert`
...
889  |     pub fn insert(&mut self, value: T) -> bool {
     |            ------ required by a bound in this associated function

error[E0277]: the trait bound `TSchemaField: std::cmp::Eq` is not satisfied
    --> /home/alex/git/volo-test/target/debug/build/volo-gen-470dfe4d935de4fb/out/volo_gen.rs:1592:29
     |
1592 |                         val.insert(<TSchemaField as ::pilota::thrift::Message>::decode_async(__protocol).await?);
     |                             ^^^^^^ the trait `std::cmp::Eq` is not implemented for `TSchemaField`
     |
note: required by a bound in `HashSet::<T, S>::insert`
    --> /home/alex/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/std/src/collections/hash/set.rs:429:8
     |
429  |     T: Eq + Hash,
     |        ^^ required by this bound in `HashSet::<T, S>::insert`
...
889  |     pub fn insert(&mut self, value: T) -> bool {
     |            ------ required by a bound in this associated function
help: consider annotating `TSchemaField` with `#[derive(Eq)]`
     |
6    +         #[derive(Eq)]
7    |         pub struct TSchemaField {
     |

error[E0277]: the trait bound `TSchemaField: Hash` is not satisfied
    --> /home/alex/git/volo-test/target/debug/build/volo-gen-470dfe4d935de4fb/out/volo_gen.rs:1592:29
     |
1592 |                         val.insert(<TSchemaField as ::pilota::thrift::Message>::decode_async(__protocol).await?);
     |                             ^^^^^^ the trait `Hash` is not implemented for `TSchemaField`
     |
note: required by a bound in `HashSet::<T, S>::insert`
    --> /home/alex/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/std/src/collections/hash/set.rs:429:13
     |
429  |     T: Eq + Hash,
     |             ^^^^ required by this bound in `HashSet::<T, S>::insert`
...
889  |     pub fn insert(&mut self, value: T) -> bool {
     |            ------ required by a bound in this associated function
```

Commenting `    6: optional map<string, list<string>> selectors;` line "solves" the issue.

```
error[E0433]: failed to resolve: could not find `pilota` in `volo_gen`
  --> src/lib.rs:12:26
   |
12 |         _name: volo_gen::pilota::FastStr,
   |                          ^^^^^^ could not find `pilota` in `volo_gen`

error[E0433]: failed to resolve: could not find `pilota` in `volo_gen`
  --> src/lib.rs:13:28
   |
13 |         _fields: volo_gen::pilota::AHashSet<::test::TSchemaField>,
   |                            ^^^^^^ could not find `pilota` in `volo_gen`

error[E0412]: cannot find type `volo_genbool` in this scope
 --> src/lib.rs:6:22
  |
6 |         _deep_check: volo_genbool,
  |                      ^^^^^^^^^^^^ not found in this scope

error[E0433]: failed to resolve: could not find `test` in the list of imported crates
  --> src/lib.rs:13:47
   |
13 |         _fields: volo_gen::pilota::AHashSet<::test::TSchemaField>,
   |                                               ^^^^ could not find `test` in the list of imported crates
   |
help: consider importing this module
   |
1  + use volo_gen::test;
   |
help: if you import `test`, refer to it directly
   |
13 -         _fields: volo_gen::pilota::AHashSet<::test::TSchemaField>,
13 +         _fields: volo_gen::pilota::AHashSet<test::TSchemaField>,
   |
```

Still there are few erros after parsing:

- ` _deep_check: volo_genbool,`
- `_exclude: volo_gen::pilota::AHashSet<::resources::Id>,` instead of `_exclude: ::pilota::AHashSet<volo_gen::resources::Id>,`
- ` _ids: volo_gen::std::vec::Vec<::resources::Id>,` instead of `_ids: ::std::vec::Vec<volo_gen::resources::Id>,`
- ```rust
    async fn get_candidates(
        &self,
        ...
        ) -> ::core::result::Result<
        ::volo_thrift::MaybeException<
            volo_gen::candidates::TCandidateList,
            CandidatesGetCandidatesException, <<< volo_gen::candidates:: is missing
        >,
        ::volo_thrift::ServerError,
    > {
        ::std::result::Result::Ok(Default::default())
    }```
- Default is not implemented for MaybeException:
```
error[E0277]: the trait bound `MaybeException<Vec<TJobInfo>, CandidatesJobInfoException>: Default` is not satisfied
   --> src/lib.rs:298:35
    |
298 |         ::std::result::Result::Ok(Default::default())
    |                                   ^^^^^^^^^^^^^^^^^^ the trait `Default` is not implemented for `MaybeException<Vec<TJobInfo>, CandidatesJobInfoException>`
```

I'll prepare a test IDLs for you.

--- a/tests/code-generation/thrift/test.thrift
+++ b/tests/code-generation/thrift/test.thrift
@@ -13,13 +13,13 @@ struct TestResponse {
 }
 
 service TestService {
-    TestResponse Test (1: TestRequest req),
+    TestResponse Test (1: TestRequest req, 2: list<TestRequest> reqs),
     TestResponse test (1: TestRequest Req),
     TestResponse Test2 (1: TestRequest type),
     TestResponse Test3 (1: TestRequest self),
 }
 
 service testService {
-    TestResponse Test (1: TestRequest req),
+    TestResponse Test (1: TestRequest req, 2: list<TestRequest> reqs),
     TestResponse test (1: TestRequest Req),
 }

--- a/volo-build/Cargo.toml
+++ b/volo-build/Cargo.toml
@@ -1,6 +1,6 @@
 [package]
 name = "volo-build"
-version = "0.10.16"
+version = "0.10.17"
 edition.workspace = true
 homepage.workspace = true
 repository.workspace = true

--- a/volo-build/src/grpc_backend.rs
+++ b/volo-build/src/grpc_backend.rs
@@ -34,7 +34,7 @@ impl VoloGrpcBackend {
     ) -> FastStr {
         let ty = self.cx().codegen_item_ty(ty.kind);
         let ty_str = if global_path {
-            format!("volo_gen{}", ty.global_path())
+            format!("{}", ty.global_path("volo_gen"))
         } else {
             format!("{}", ty)
         };
@@ -54,7 +54,7 @@ impl VoloGrpcBackend {
     ) -> FastStr {
         let ret_ty = self.cx().codegen_item_ty(ty.kind);
         let ret_ty_str = if global_path {
-            format!("volo_gen{}", ret_ty.global_path())
+            format!("{}", ret_ty.global_path("volo_gen"))
         } else {
             format!("{}", ret_ty)
         };
@@ -402,7 +402,7 @@ impl CodegenBackend for VoloGrpcBackend {
             "",
             |enum_variant_names| -> "Self::{enum_variant_names}(s) => {{
                 ::volo_grpc::codec::encode::encode(s, compression_encoding)
-            }}," 
+            }},"
         );
 
         let req_recv_from_body = crate::join_multi_strs!(

--- a/volo-build/src/thrift_backend.rs
+++ b/volo-build/src/thrift_backend.rs
@@ -4,7 +4,6 @@ use pilota_build::{
     db::RirDatabase,
     rir::{self, Method},
     tags::RustWrapperArc,
-    ty::TyKind,
     CodegenBackend, Context, DefId, IdentName, Symbol, ThriftBackend,
 };
 use quote::format_ident;
@@ -648,11 +647,8 @@ impl pilota_build::CodegenBackend for VoloThriftBackend {
         let mut ret_ty = self
             .inner
             .codegen_item_ty(method.ret.kind.clone())
-            .global_path()
+            .global_path("volo_gen")
             .to_string();
-        if need_prepend_volo_gen_path(&method.ret.kind) {
-            ret_ty = format!("volo_gen{ret_ty}");
-        }
         if let Some(RustWrapperArc(true)) = self
             .cx()
             .tags(method.ret.tags_id)
@@ -665,13 +661,12 @@ impl pilota_build::CodegenBackend for VoloThriftBackend {
             .args
             .iter()
             .map(|a| {
-                let ty = self.inner.codegen_item_ty(a.ty.kind.clone()).global_path();
+                let ty = self
+                    .inner
+                    .codegen_item_ty(a.ty.kind.clone())
+                    .global_path("volo_gen");
                 let ident = self.cx().rust_name(a.def_id).0.field_ident(); // use the _{rust-style fieldname} without keyword escaping
-                if need_prepend_volo_gen_path(&a.ty.kind) {
-                    format!("_{ident}: volo_gen{ty}")
-                } else {
-                    format!("_{ident}: {ty}")
-                }
+                format!("_{ident}: {ty}")
             })
             .join(",");
 
@@ -698,14 +693,6 @@ impl pilota_build::CodegenBackend for VoloThriftBackend {
 
     fn cx(&self) -> &Context {
         self.inner.cx()
-    }
-}
-
-fn need_prepend_volo_gen_path(ty: &TyKind) -> bool {
-    match ty {
-        TyKind::Arc(t) => need_prepend_volo_gen_path(&t.kind),
-        TyKind::Path(_) => true,
-        _ => false,
     }
 }
 

--- a/volo-thrift/Cargo.toml
+++ b/volo-thrift/Cargo.toml
@@ -1,6 +1,6 @@
 [package]
 name = "volo-thrift"
-version = "0.10.4"
+version = "0.10.5"
 edition.workspace = true
 homepage.workspace = true
 repository.workspace = true

--- a/volo-thrift/src/error.rs
+++ b/volo-thrift/src/error.rs
@@ -229,3 +229,9 @@ pub enum MaybeException<T, E> {
     Ok(T),
     Exception(E),
 }
+
+impl<T: Default, E> Default for MaybeException<T, E> {
+    fn default() -> Self {
+        MaybeException::Ok(Default::default())
+    }
+}

